src/server_manager/web_app/app.spec.ts

```diff
@@ -64,13 +233,24 @@ function createTestApp(
   if (!manualServerRepo) {
     manualServerRepo = new FakeManualServerRepository();
   }
-  const fakeDisplayServerRepository = new FakeDisplayServerRepository();
+  if (!displayServerRepository) {
+    displayServerRepository = new FakeDisplayServerRepository();
+  }
   return new App(
       polymerAppRoot, WEB_APP_URL, VERSION, fakeDigitalOceanSessionFactory,
-      fakeDigitalOceanServerRepositoryFactory, manualServerRepo, fakeDisplayServerRepository,
+      fakeDigitalOceanServerRepositoryFactory, manualServerRepo, displayServerRepository,
       digitalOceanTokenManager);
 }
 
+async function makeDisplayServer(server: server.Server) {
+  return {
```

Comment:
This is a near duplicate of code in `app.ts`. I wonder if it's worth factoring that out to a function. It maybe best-placed in `display}_server.ts`.

Reply:
Good catch, I've implemented your suggestion in f0f930b.